charts/reconstruction-server/templates/deployment.yaml

```
@@ -44,10 +44,22 @@ spec:
               containerPort: {{ int $portSpec.port }}
               protocol: {{ $portSpec.protocol }}
             {{- end }}
+          {{- if .Values.resources }}
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
+          {{- end }}
+          {{- if .Values.envFrom }}
+          envFrom:
+            {{- toYaml .Values.envFrom | nindent 12 }}
+          {{- end }}
+          {{- $env := list }}
+          {{- range .Values.extraEnv }}
+          {{- $env = append $env . }}
+          {{- end }}
+          {{- if $env }}
           env:
-            {{- toYaml .Values.extraEnv | nindent 12 }}
+            {{- toYaml $env | nindent 12 }}
+          {{- end }}
```

I think it would be nice if we could use `with` instead of `if` like the rest of the template sections

```suggestion
          {{- with .Values.resources }}
          resources:
            {{- toYaml . | nindent 12 }}
          {{- end }}
          {{- with .Values.envFrom }}
          envFrom:
            {{- toYaml . | nindent 12 }}
          {{- end }}
          {{- $env := list }}
          {{- range .Values.extraEnv }}
          {{- $env = append $env . }}
          {{- end }}
          {{- if $env }}
          env:
            {{- toYaml $env | nindent 12 }}
          {{- end }}
```